pvlib/pvsystem.py

```diff
@@ -1679,12 +1679,12 @@ def sapm(effective_irradiance, temp_cell, module):
     # TODO: remove this warning in v0.8 after deprecation period for change in
     # effective irradiance units, made in v0.7
     with np.errstate(invalid='ignore'):  # turn off warning for NaN
-        if np.all(
-            effective_irradiance[np.where(effective_irradiance > 0.0)] < 2.0):
-                import warnings
-                warnings.warn('effective_irradiance inputs appear to be in',
-                              ' suns. Units changed in v0.7 from suns to W/m2',
-                              RuntimeWarning)
+        ee = effective_irradiance
+        if np.all(ee[ee > 0.0] < 2.0):
```

Comment:
```suggestion
        if np.all(np.logical_and(ee > 0.0, ee < 2.0)):
```

Reply:
this should preserve compatibility with scalar input

Reply:
Not to continue complicating this simple warning, but `np.all(np.logical_and(ee > 0.0, ee < 2.0))` will return False if ANY zeros are present, which doesn't seem desirable.  Would using `ee = np.asarray(effective_irradiance)` with Cliff's code work?

Reply:
How about:
```(np.nanmin(ee) > 0.0) and (np.nanmax(ee) < 2.0)```

Reply:
Thanks for the helpful suggestions.